src/components/organisms/topbar/dashboard.tsx

```diff
@@ -39,23 +52,141 @@ export const DashboardTopbar = () => {
 		});
 	};
 
+	useEffect(() => {
+		const getAndSaveFiles = async () => {
+			if (!projectId || !templateFiles) return;
+
+			const fileEntries = Object.entries(templateFiles).map(([path, fileNode]) => {
+				const content = "content" in fileNode ? fileNode.content : "";
+
+				return [path, new Uint8Array(new TextEncoder().encode(content))];
+			});
+
+			await saveAllFiles(Object.fromEntries(fileEntries));
+			addToast({
+				message: t("projectCreatedSuccessfully"),
+				type: "success",
+			});
+
+			getProjectsList();
+			navigate(`/projects/${projectId}`);
+		};
+
+		getAndSaveFiles();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [projectId]);
+
+	const handleImportFile = async (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Let's take this whole function to the `useProjectStore`

Reply:
added to new hook and optimization creating a new project